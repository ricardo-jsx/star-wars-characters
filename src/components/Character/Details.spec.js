import React from 'react';
import { render } from '@testing-library/react';

import Details from './Details';

describe('components/Character/Details', () => {
  it('should display all character infos', () => {
    const characterMock = {
      id: 1,
      name: 'Luke Skywalker',
      height: '172',
      mass: '77',
      hair_color: 'blond',
      birth_year: '19BBY',
      gender: 'male',
      skin_color: 'fair',
      img: '/resources/characters/1.jpg',
      starships: [
        {
          id: 12,
          name: 'X-wing',
          img: '/resources/starships/12.jpg',
        },
        {
          id: 22,
          name: 'Imperial shuttle',
          img: '/resources/starships/22.jpg',
        },
      ],
    };

    const { container } = render(<Details character={characterMock} />);
    expect(container).toMatchSnapshot();
  });
});
