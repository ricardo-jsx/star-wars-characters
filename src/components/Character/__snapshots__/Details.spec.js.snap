// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Character/Details should display all character infos 1`] = `
<div>
  <section
    class="sc-AxjAm hkfvXl"
  >
    <h1>
      Character Details
    </h1>
    <div>
      <span>
        Name
      </span>
      <span>
        Luke Skywalker
      </span>
    </div>
    <div>
      <span>
        Height
      </span>
      <span>
        172
      </span>
    </div>
    <div>
      <span>
        Mass
      </span>
      <span>
        77
         KG
      </span>
    </div>
    <div>
      <span>
        Hair Color
      </span>
      <span>
        blond
      </span>
    </div>
    <div>
      <span>
        Birth Year
      </span>
      <span>
        19BBY
      </span>
    </div>
    <div>
      <span>
        Gender
      </span>
      <span>
        male
      </span>
    </div>
    <div>
      <span>
        Skin Color
      </span>
      <span>
        fair
      </span>
    </div>
    <div>
      <span>
        Starships
      </span>
      <span>
        X-wing, Imperial shuttle
      </span>
    </div>
  </section>
</div>
`;
